bndtools.core/src/bndtools/editor/project/RunRequirementsPart.java

```diff
@@ -257,8 +259,8 @@ private void doAddBundle() {
             WizardDialog dialog = new WizardDialog(getSection().getShell(), wizard);
 
             if (Window.OK == dialog.open()) {
-                List<VersionedClause> result = wizard.getSelectedBundles();
-                List<Requirement> adding = new ArrayList<Requirement>(result.size());
+                final List<VersionedClause> result = wizard.getSelectedBundles();
+                final List<Requirement> adding = new ArrayList<Requirement>(result.size());
```

Comment:
Just do `adding.removeAll(requires)` to remove any existing requirements before the empty check. Then the editor will only be marked dirty if there are new requirements being added.